modules/images/webp-uploads/load.php

```diff
@@ -704,3 +704,187 @@ function webp_uploads_update_rest_attachment( WP_REST_Response $response, WP_Pos
+
+	foreach ( $trace as $element ) {
+		if ( ! isset( $element['function'] ) ) {
+			continue;
+		}
+
+		switch ( $element['function'] ) {
+			case 'wp_save_image':
+				// Right after an image has been edited.
+				return webp_uploads_backup_sources( $attachment_id, $data );
+			case 'wp_restore_image':
+				// When an image has been restored.
+				return webp_uploads_restore_image( $attachment_id, $data );
+		}
+	}
+
+	return $data;
+}
+add_filter( 'wp_update_attachment_metadata', 'webp_uploads_update_attachment_metadata', 10, 2 );
+
+/**
+ * Before saving the metadata of the image store a backup values for the sources and file property
+ * those files would be used and deleted by the backup mechanism, right after the metadata has
+ * been updated. It removes the current sources property due once this function is executed
+ * right after an edit has taken place and the current sources are no longer accurate.
+ *
+ * @since n.e.x.t
+ *
+ * @param int   $attachment_id The ID representing the attachment.
+ * @param array $data The current metadata of the attachment.
+ * @return array The updated metadata for the attachment.
+ */
+function webp_uploads_backup_sources( $attachment_id, $data ) {
+	$metadata = wp_get_attachment_metadata( $attachment_id );
+	// Nothing to back up.
+	if ( ! isset( $metadata['sources'] ) ) {
+		return $data;
+	}
+
+	$target = isset( $_REQUEST['target'] ) ? sanitize_text_field( $_REQUEST['target'] ) : 'all';
+
+	// When an edit to an image is only applied to a thumbnail there's nothing we need to back up.
+	if ( 'thumbnail' === $target ) {
+		return $data;
+	}
+
+	$sources = $metadata['sources'];
+	// Prevent execution of the callbacks more than once if the callback was already executed.
+	$executed = false;
+
+	add_action(
+		'added_post_meta',
+		function ( $meta_id, $attachment_id, $meta_name, $backup_sizes ) use ( $sources, &$executed ) {
+			// The backup sources array.
+			if ( '_wp_attachment_backup_sizes' !== $meta_name || $executed ) {
+				return;
+			}
+
+			$executed = true;
+			webp_uploads_backup_full_image_sources( $attachment_id, $sources );
+		},
+		10,
+		4
+	);
+
+	add_action(
+		'updated_post_meta',
+		function ( $meta_id, $attachment_id, $meta_name, $backup_sizes ) use ( $sources, &$executed ) {
+			// The backup sources array.
+			if ( '_wp_attachment_backup_sizes' !== $meta_name || $executed ) {
+				return;
+			}
+			$executed = true;
+			webp_uploads_backup_full_image_sources( $attachment_id, $sources );
+		},
+		10,
+		4
+	);
+
+	// Remove the current sources as at this point the current values are no longer accurate.
+	// TODO: Requires to be updated from https://github.com/WordPress/performance/issues/158.
+	unset( $data['sources'] );
+
+	return $data;
+}
+
+/**
+ * Stores the provided sources for the attachment ID in the `_wp_attachment_backup_sources`  with
+ * the next available target if target is `null` no source would be stored.
+ *
```

Comment:
Remember to provide `@since` annotations.
```suggestion
 * the next available target if target is `null` no source would be stored.
 *
 * @since n.e.x.t
 *
```